### PR TITLE
Mark issues and pull requests as stale

### LINKED ISSUE #6610: Fix bot marking issues as stale and rotten
There used to be automatization that marked issues as stale and then automatically closed them as rotten. 
It looks like this is not working anymore. We should fix it.

### PR DESCRIPTION
**Pull request by ritudes**
**What type of PR is this:**

**What does this PR do / why we need it:**

**Which issue(s) this PR fixes:**

Fixes #6610 

**PR acceptance criteria:**

- [ ] Unit test 

- [ ] Integration test 

- [ ] Documentation 

**How to test changes / Special notes to the reviewer:**

### CODE REVIEW

File: .github/workflows/stale.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      pull-requests: write  # for actions/stale to close stale PRs
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/stale@v4.1.1
+      with:
+        repo-token: ${{ secrets.GITHUB_TOKEN }}
+        stale-issue-message: 'A friendly reminder that this issue had no activity for 30 days.'
+        stale-pr-message: 'A friendly reminder that this PR had no activity for 30 days.'
+        stale-issue-label: 'stale-issue'
```
Comment (rm3l) [inline .github/workflows/stale.yaml:27]:
```suggestion
        stale-issue-label: 'lifecycle/stale'
        close-issue-label: 'lifecycle/rotten'
        exempt-issue-labels: 'lifecycle/active,lifecycle/frozen'
```

File: .github/workflows/stale.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        stale-pr-label: 'stale-pr'
```
Comment (rm3l) [inline .github/workflows/stale.yaml:28]:
```suggestion
        stale-pr-label: 'lifecycle/stale'
        close-pr-label: 'lifecycle/rotten'
        exempt-pr-labels: 'lifecycle/active,lifecycle/frozen'
```

File: .github/workflows/stale.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      issues: write  # for actions/stale to close stale issues
+      pull-requests: write  # for actions/stale to close stale PRs
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/stale@v4.1.1
+      with:
+        repo-token: ${{ secrets.GITHUB_TOKEN }}
+        stale-issue-message: 'A friendly reminder that this issue had no activity for 30 days.'
+        stale-pr-message: 'A friendly reminder that this PR had no activity for 30 days.'
```
Comment (rm3l) [inline .github/workflows/stale.yaml:26]:
```suggestion
        stale-pr-message: 'A friendly reminder that this PR had no activity for 90 days. Stale PRs will be closed after an additional 30 days of inactivity.'
```

File: .github/workflows/stale.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        stale-issue-label: 'stale-issue'
+        stale-pr-label: 'stale-pr'
+        days-before-stale: 30
```
Comment (rm3l) [inline .github/workflows/stale.yaml:29]:
```suggestion
        days-before-stale: 90
```

**Comment by ritudes** [comment]
Thank you for your reviews @rm3l .I have applied those changes as per your suggestions

**Review by rm3l — APPROVED** [review]

**Comment by rm3l** [comment]
```
  [FAILED] registry_list_output.mdx
  Expected
      <string>:   strings.Join({
        	... // 108 identical bytes
        	"                        VERSIONS\nnodejs-react  StagingRegistry  ",
        	"       React is a free and open-source front-en...  2.0.2, 2.1.0",
      + 	", 2.2.0",
        	"\nnodejs-react  DefaultDevfileRegistry  React is a free and open-",
        	"source front-en...  2.0.2, 2.1.0\n```\n",
        }, "")
      
  to be empty
  In [It] at: /go/odo_1/tests/documentation/command-reference/doc_command_reference_init_test.go:189 @ 06/09/23 12:49:03.116
------------------------------
â€¢â€¢â€¢â€¢â€¢â€¢Spawning '/go/bin/odo init' from /tmp/591808707
â€¢â€¢â€¢Spawning '/go/bin/odo delete project odo-dev' from /tmp/2208322380
â€¢â€¢â€¢â€¢

Summarizing 1 Failure:
  [FAIL] doc command reference odo init Non Interactive Mode fetching devfile from a registry when setting up the registry [It] Fetch Devfile from any registry of the list [nocluster]
  /go/odo_1/tests/documentation/command-reference/doc_command_reference_init_test.go:189

Ran 16 of 16 Specs in 46.762 seconds
FAIL! -- 15 Passed | 1 Failed | 0 Pending | 0 Skipped
``` 

Reported in https://github.com/redhat-developer/odo/issues/6876

/override Kubernetes-Integration-Tests/Kubernetes-Docs-Integration-Tests

**Comment by rm3l** [comment]
```
...
 [odo]  âš   AttachVolume.Attach failed for volume "pvc-76ffbf80-442b-4a64-825b-8b931bde3264" : timed out waiting for external-attacher of vpc.block.csi.ibm.io CSI driver to attach volume r010-796cee09-5495-42c5-9c28-f0c09c729b4a
  [odo]  âš   Unable to attach or mount volumes: unmounted volumes=[odo-projects], unattached volumes=[odo-projects odo-shared-data kube-api-access-5hsp9]: timed out waiting for the condition
  [FAILED] in [BeforeEach] - /go/odo_1/tests/helper/helper_run.go:54 @ 06/09/23 12:33:41.225
  Deleting project: cmd-dev-test468xck
...
  [odo]  âš   AttachVolume.Attach failed for volume "pvc-dbbed17f-7a30-40d4-b0a5-ac4470f8ae04" : timed out waiting for external-attacher of vpc.block.csi.ibm.io CSI driver to attach volume r010-e0f0bcbe-af9b-45ae-8a75-4ecf629092ef
  [odo]  âš   AttachVolume.Attach failed for volume "pvc-a48b435e-ac81-498e-9cd1-fdc4f9e6b79a" : timed out waiting for external-attacher of vpc.block.csi.ibm.io CSI driver to attach volume r010-61568a66-adc6-417b-89d0-d0bc25dc8449
  [odo]  âš   Unable to attach or mount volumes: unmounted volumes=[firstvol-araele-app-vol secondvol-araele-app-vol], unattached volumes=[odo-projects odo-shared-data firstvol-araele-app-vol secondvol-araele-app-vol kube-api-access-jj9qx]: timed out waiting for the condition
...

Summarizing 2 Failures:
  [FAIL] odo dev command tests when running odo dev with devfile containing volume-component - without metadata.name [BeforeEach] should successfully use the volume components in container components
  /go/odo_1/tests/helper/helper_run.go:54
  [FAIL] odo dev command tests when a component is bootstrapped when odo dev is executed and Ephemeral is set to false [BeforeEach] when killing odo dev and running odo delete component --wait should have deleted all resources before returning
  /go/odo_1/tests/helper/helper_run.go:54

Ran 489 of 932 Specs in 1569.096 seconds
FAIL! -- 487 Passed | 2 Failed | 0 Pending | 443 Skipped
```

Issues with attaching volumes on OpenShift.
Overriding because the changes here are only about adding a new GitHub Workflow for housekeeping..

/override OpenShift-Integration-tests/OpenShift-Integration-tests

**Pull request merged**